front/src/modules/simulationResult/components/SpaceTimeChart/useTrackOccupancy.ts

```diff
@@ -354,38 +357,49 @@ const useTrackOccupancy = ({
     const pathOperationalPointsWithoutTracks = pathOperationalPoints.filter(
       (op) => !(tracksState.data || {})[op.waypointId]
     );
-    const loadAllTracks = async (
-      operationalPointReferences: {
-        operational_point: string;
-      }[]
-    ) => {
+    const loadAllTracks = async (operationalPointReferences: { operational_point: string }[]) => {
       setTracksState((state) => ({ type: 'loading', data: state.data || {} }));
 
       try {
         const data = await postInfraByInfraIdMatchOperationalPoints({
           infraId,
-          body: {
-            operational_point_references: operationalPointReferences,
-          },
+          body: { operational_point_references: operationalPointReferences },
         }).unwrap();
 
         if (aborted) return;
 
-        setTracksState({
-          type: 'ok',
-          data: fromPairs(
-            operationalPointReferences.map(({ operational_point }, i) => [
-              operational_point,
-              uniqBy(
-                data.related_operational_points[i][0].parts.map((part) => ({
+        const allTrackIds = data.related_operational_points.flatMap(([points]) =>
+          points.parts.map((part) => part.track)
+        );
+        const fetchedTrackSections = await getTrackSectionsByIds(allTrackIds);
+
+        const trackSectionByTrackId = new Map();
+        for (const trackSections of Object.values(fetchedTrackSections)) {
+          if (trackSections.id) trackSectionByTrackId.set(trackSections.id, trackSections);
+          trackSectionByTrackId.set(trackSections.id, trackSections);
```

Comment:
These 2 lines are the same and the `if` is not necessary, `trackSections.id` is always defined.

```suggestion
          trackSectionByTrackId.set(trackSections.id, trackSections);
```